apps/launch/components/MyStakingContracts/Details.tsx

```diff
@@ -1,13 +1,248 @@
+  flex-direction: column;
+  max-width: 720px;
+  margin: 0 auto;
+`;
+
+const Container: FC<{ children: ReactNode }> = ({ children }) => (
+  <StyledMain>
+    <Card>{children}</Card>
+  </StyledMain>
+);
+
+const Loader = () => (
+  <Container>
+    <Flex justify="center" align="center" style={{ height: '300px' }}>
+      <Spin />
+    </Flex>
+  </Container>
+);
+
+const NotFound = () => {
+  const router = useRouter();
+  const { networkName } = useAppSelector((state) => state.network);
+
+  return (
+    <Container>
+      <Title level={4} className="mt-0">
+        Contract not found...
+      </Title>
+      <Paragraph>
+        We couldn’t find the staking contract you’re referring to. Go to your staking contracts and
+        try again.
+      </Paragraph>
+
+      <Flex justify="center">
+        <Button type="primary" onClick={() => router.push(`/${networkName}/my-staking-contracts`)}>
+          My staking contracts
+        </Button>
+      </Flex>
+    </Container>
+  );
+};
+
+type ColFlexContainerProps = { text: string; content: ReactNode };
+const ColFlexContainer: FC<ColFlexContainerProps> = ({ text, content }) => {
+  return (
+    <Col span={12}>
+      <Flex vertical gap={4} align="flex-start">
+        <Text type="secondary">{text}</Text>
+        {content}
+      </Flex>
+    </Col>
+  );
+};
+
+const NominatedForIncentives: FC<{ isNominated: boolean }> = ({ isNominated }) => {
+  const router = useRouter();
+  const { networkName } = useAppSelector((state) => state.network);
+
+  return isNominated ? (
+    <Tag color="success">
+      <CheckCircleOutlined />
+      &nbsp;Nominated
+    </Tag>
+  ) : (
+    <Button type="primary" onClick={() => router.push(`/${networkName}/${URL.nominateContract}`)}>
+      Nominate
+    </Button>
+  );
+};
+
+const ContractAddress: FC<{ address: string }> = ({ address }) => {
+  const { networkId } = useAppSelector((state) => state.network);
+  const suffixCount = 5;
+  const start = address.slice(0, address.length - suffixCount);
+  const suffix = address.slice(-suffixCount).trim();
+
+  if (!networkId) return null;
+
+  return (
+    <Flex align="flex-start" gap={4}>
+      <Link
+        href={`${EXPLORER_URLS[networkId]}/address/${address}`}
+        target="_blank"
+        rel="noreferrer"
+      >
+        <Text style={{ maxWidth: '140px', color: COLOR.PRIMARY }} ellipsis={{ suffix }}>
+          {start}
+        </Text>
+        &nbsp;↗
+      </Link>
+
+      <Button
+        size="small"
+        icon={<CopyOutlined />}
+        onClick={() => navigator.clipboard.writeText(address)}
+        style={{ margin: '-4px 0 0 8px' }}
+      />
+    </Flex>
+  );
+};
+
+const Template: FC<{ template: string }> = ({ template }) => {
+  const { networkId } = useAppSelector((state) => state.network);
+
+  return (
+    <Flex vertical gap={4}>
+      <Text>{template}</Text>
+      {networkId ? (
+        <Link href={EXPLORER_URLS[networkId]} target="_blank" rel="noreferrer">
```

Comment:
this link doesn't have address, can you please add; and can we just use an "a" tag? so that we probably won't need Text with additional styles